pkg/kyverno/test/test_command.go

```diff
@@ -577,32 +599,61 @@ func applyPoliciesFromPath(fs billy.Filesystem, policyBytes []byte, valuesFile s
 	engineResponses := make([]*response.EngineResponse, 0)
 	var dClient *client.Client
 	values := &Test{}
+	versionedvalues := &versionedTest{}
+	versioned := true
 	var variablesString string
 	var pvInfos []policyreport.Info
 	var resultCounts common.ResultCounts
+	var resultsMap map[string]report.PolicyReportResult
+	var testResults []TestResults
+	var policyFullPath []string
+	var resourceFullPath []string
 	store.SetMock(true)
 
-	if err := json.Unmarshal(policyBytes, values); err != nil {
+	if err := json.Unmarshal(policyBytes, versionedvalues); err != nil {
 		return sanitizederror.NewWithError("failed to decode yaml", err)
 	}
+	if versionedvalues.APIVersion == "" {
+		versioned = false
+		if err := json.Unmarshal(policyBytes, values); err != nil {
+			return sanitizederror.NewWithError("failed to decode yaml", err)
+		}
+	}
+
+	log.Log.V(5).Info("valuesFile = ", valuesFile)
+
+	if versioned {
```

Comment:
I think overall implementation could be much cleaner if this was split into two functions. One functions only handles the legacy format, one function only handles the new format. 

That way when we get rid of the old format eventually, we just remove the function. This here is going to cause hard to read code.

Reply:
I could change this if decided previously I felt that it made the code bigger, but if you suggest I will change this, and for the other part; when we need to get rid of the old format we could remove the else part of these blocks
```
if versioned {
--} else {
---
}
```